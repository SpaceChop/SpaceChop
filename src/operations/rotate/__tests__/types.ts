import RotateConfig, { RotateConfig as Config } from '../types';

describe('should validate rotate config', () => {
  describe('should throw on invalid config', () => {
    it('require angle', () => {
      const config = {};
      expect(() => RotateConfig.check(config)).toThrowErrorMatchingSnapshot();
    });

    it('require valid width and height', () => {
      let config: Config = {
        angle: -361,
      };
      expect(() => RotateConfig.check(config)).toThrowErrorMatchingSnapshot();

      config = {
        angle: 361,
      };
      expect(() => RotateConfig.check(config)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('should now throw on valid config', () => {
    it('width & height', () => {
      let config: Config = {
        angle: 90,
      };
      expect(() => RotateConfig.check(config)).not.toThrow();

      config = {
        angle: 'auto',
      };
      expect(() => RotateConfig.check(config)).not.toThrow();
    });
  });
});
