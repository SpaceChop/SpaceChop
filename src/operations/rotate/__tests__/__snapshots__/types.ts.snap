// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should validate rotate config should throw on invalid config require angle 1`] = `"Expected number | \\"auto\\" | { from_path: string; }, but was undefined"`;

exports[`should validate rotate config should throw on invalid config require valid width and height 1`] = `"Expected number | \\"auto\\" | { from_path: string; }, but was number"`;

exports[`should validate rotate config should throw on invalid config require valid width and height 2`] = `"Expected number | \\"auto\\" | { from_path: string; }, but was number"`;
